### ISSUE TITLE
Not opening chest in Ancient Tunnels

### ISSUE THREAD

**Issue by pananana** [kolton/d2bot-with-kolbot#2711]
Scripts.AncientTunnels = true;
		Config.AncientTunnels.OpenChest = true; // Open special chest in Lost City
		Config.AncientTunnels.KillDarkElder = false;

I have sat and watched the bot kill all the mobs around the chest and then TP without even touching it, any idea what could be conflicting with it?

EDIT: Have realised that the config is for the chest in Lost City. Is there a way for me to make it open the chest inside the Ancient Tunnels?

**Comment by DarkHorseDre** [comment]
Do you have `Config.OpenChests = true;` set?

**Comment by pananana** [comment]
> Do you have `Config.OpenChests = true;` set?

Yeah I have done now and it is opening the chest, but it also opening chests on every script being run within the game, is there a way for me to specifically only open the chest on the ancient tunnels script?

**Comment by DarkHorseDre** [comment]
There are many ways to do it - you could enable `openchests` for this script only (I think there's only 1 chest there anyway?) or call the script with `openchests=true` as a param, or just open it specifically like the lost city chest in the AT script.

to do the latter, just add the last if block from this: https://pastebin.com/1gpJcVWQ

I've not taken the time to add a config parameter to enable you to control this from the char config, but you can copy the method used for `Config.AncientTunnels.OpenChest` and update `config.js` and your char config accordingly.

I'd also suggest you read the entire char config and wiki guide as there's lots of cool stuff in there: https://github.com/kolton/d2bot-with-kolbot/wiki/Kolbot-Character-config

**Comment by pananana** [comment]
> There are many ways to do it - you could enable `openchests` for this script only (I think there's only 1 chest there anyway?) or call the script with `openchests=false` as a param, or just open it specifically like the lost city chest in the AT script.
> 
> to do the latter, just add the last if block from this: https://pastebin.com/1gpJcVWQ
> 
> I've not taken the time to add a config parameter to enable you to control this from the char config, but you can copy the method used for `Config.AncientTunnels.OpenChest` and update `config.js` and your char config accordingly.
> 
> I'd also suggest you read the entire char config and wiki guide as there's lots of cool stuff in there: https://github.com/kolton/d2bot-with-kolbot/wiki/Kolbot-Character-config

Perfect, thanks for the help, it's running exactly as I want it now.

**Comment by DarkHorseDre** [comment]
coolio

I had another thought - if config.openchests is true, this has no effect.
so you'd probably only use this if it is false, so you could put a condition on it, like:

```js
if (!Config.OpenChests && Pather.moveToPreset(me.area, 2, 397) && Misc.openChests(5)) {
        Pickit.pickItems();
    }
```

**Comment by Samael36** [comment]
Hey guys, thanks for the solution! It worked for me in Ancient Tunnels, but I'd also like to add it to other areas such as the Mausoleum.
How can I find the numbers to change here: (me.area, 2, 397) && Misc.openChests(5) ?
